### PR TITLE
xmacro.ini

### PR DESCRIPTION
**Pull request by linuxCowboy**
Define a fix init path / file.

default: hidden file in homedir

### CODE REVIEW

File: config1.c [hunk 1]
```diff
@@ -194,19 +194,31 @@ static void cfg_open_parse_file(config_obj *cfgo, FILE *fp)
 config_obj *cfg_open(char *url)
 {
 	config_obj *cfgo = NULL;
+        char *initpath = INIT_PATH;
+        char initfile[400];
+
```
Comment (CyberShadow) [inline config1.c:199]: Strange whitespace (spaces instead of tabs?)

**Review by CyberShadow — COMMENTED** [review]
I think we should only look in a new location if there is no configuration at the old location. That way if someone were to upgrade the program to this version, things wouldn't stop working because it would no longer read the configuration file from the old location.

Also I think for new code it is better to use XDG conventions (i.e. `~/.config/xmacro.ini` OSLT)?

File: config1.c (hunk 1 above)
Comment (linuxCowboy) [reply config1.c:199]: Yes, maybe. But it's better than creating a config file in every directory where it happens to run. _That_ bothers me in the first place.

If you check and read an old file, write it back to this old place or in a fix user dir or a system dir? Levels like Systemd?

It's messy and little to gain.

And I can't see why I need a _ini_ at all?

My first patch was a simple


```
 static void cfg_save(config_obj *cfgo)
 {
-	if(cfgo == NULL)
+	if(1 || cfgo == NULL)
 	{
 		return;
 	}

```
in **cfg_save**(), and it seems enough...

Comment (CyberShadow) [reply config1.c:199]: > If you check and read an old file, write it back to this old place or in a fix user dir or a system dir? Levels like Systemd?

Oh, I missed that it writes back to it. This program is weird.

> And I can't see why I need a _ini_ at all?

I don't know either.

Maybe the location of the configuration file should be read from an environment variable, or a command line option?

I don't think it makes sense to have the program silently write to a global (per-user) configuration file.

**Pull request closed without merging**